### PR TITLE
Port vertical changes

### PR DESCRIPTION
**Pull request by prashanth-intel**

### CODE REVIEW

File: scripts/setup_civ_ini.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# Copyright (c) 2022 Intel Corporation.
+# All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -eE
+CIV_WORK_DIR=$(pwd)
+USER_HOME=$(getent passwd $SUDO_USER | cut -d: -f6)
+
+function setup_civ_ini() {
```
Comment (YadongQi) [inline scripts/setup_civ_ini.sh:12]: I think its better to keep this script file in vertical.

**Comment by YadongQi** [comment]
The S branch has been into maintaince status, features are not accepted in genneral unless there is a strong requirement. Please make sure you really want these new updates.
And please also fix the code scaning issues.

**Comment by prashanth-intel** [comment]
> The S branch has been into maintaince status, features are not accepted in genneral unless there is a strong requirement. Please make sure you really want these new updates. And please also fix the code scaning issues.

Hi Yadong. Removed the commit for ini file. Fixed all relevant code scanning issues. There is one issue which does not seem critical as I am appending an integer and not string.

File: scripts/setup_host.sh [hunk 2]
```diff
@@ -667,6 +667,8 @@ ubu_enable_host_sriov
 ubu_update_fw
 
 install_vm_manager
+create_aaf_dir
+setup_civ_ini
```
Comment (YadongQi) [inline scripts/setup_host.sh:671]: calling inexistent function?

**Comment by YadongQi** [comment]
> > The S branch has been into maintaince status, features are not accepted in genneral unless there is a strong requirement. Please make sure you really want these new updates. And please also fix the code scaning issues.
> 
> Hi Yadong. Removed the commit for ini file. Fixed all relevant code scanning issues. There is one issue which does not seem critical as I am appending an integer and not string.

Suggest to use strncat() to avoid unsafe use of strcat().

**Comment by prashanth-intel** [comment]
> > > The S branch has been into maintaince status, features are not accepted in genneral unless there is a strong requirement. Please make sure you really want these new updates. And please also fix the code scaning issues.
> > 
> > 
> > Hi Yadong. Removed the commit for ini file. Fixed all relevant code scanning issues. There is one issue which does not seem critical as I am appending an integer and not string.
> 
> Suggest to use strncat() to avoid unsafe use of strcat().

Done. Thanks for the suggestion.

File: src/guest/start.c [hunk 3]
```diff
@@ -856,6 +908,25 @@ static void strip_duplicate(gchar *val, const gchar *inner_cmd)
...
 }
 
+void set_pwr_ctrl(char * qmp_cmd) {
+	for(int avail = 0; avail < MAX_NUM_GUEST; avail++) {
+		char *num;
+		asprintf(&num, "%d", avail);
+		strcpy(qmp_pwr_socket, "/tmp/qmp-pwr-socket-");
+		if(sizeof(qmp_pwr_socket) > 0) {
+			strncat(qmp_pwr_socket, num, sizeof(int));
+		}
```
Comment (YadongQi) [inline src/guest/start.c:918]: sizeof(qmp_pwr_socket) always > 0, so the condition always true. I think you mean to:
```
if (strlen(num) < (sizeof(qmp_pwr_socket) - strlen(qmp_pwr_socket)) {
    strncat(qmp_pwr_socket, num, (sizeof(qmp_pwr_socket) - strlen(qmp_pwr_socket));
}
```

File: src/guest/start.c [hunk 4]
```diff
@@ -906,6 +977,11 @@ int start_guest(char *name)
 	cx = snprintf(p, size, " -qmp unix:%s/.%s"CIV_GUEST_QMP_SUFFIX",server,nowait", civ_config_path, qname);
 	p += cx; size -= cx;
 
+	const char qmp_sock[MAX_CMDLINE_LEN] = { 0 };
```
Comment (YadongQi) [inline src/guest/start.c:980]: remove `const ` qualifier.

File: src/guest/start.c [hunk 5]
```diff
@@ -856,6 +908,25 @@ static void strip_duplicate(gchar *val, const gchar *inner_cmd)
 		g_free(to_split);
 }
 
+void set_pwr_ctrl(char * qmp_cmd) {
```
Comment (YadongQi) [inline src/guest/start.c:911]: add `static` qualifier for this function.

File: src/guest/start.c [hunk 6]
```diff
@@ -552,6 +596,13 @@ static void cleanup_passthrough(void) {
 	}
 }
 
+void cleanup_pwr_ctrl() {
```
Comment (YadongQi) [inline src/guest/start.c:599]: add `static` qualifier for this function.

**Review by YadongQi — CHANGES_REQUESTED** [review]

**Review by YadongQi — APPROVED** [review]

**Pull request merged**